expences-app-swiftui/Views/ExpenseEditingView.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  AddExpense.swift
+//  expences-app-swiftui
+//
+//  Created by Serhii Prykhodko on 16.10.2020.
+//
+
+import SwiftUI
+
+struct ExpenseEditingView: View {
+    
+    @Binding var expense: Expense?
+    @State var name = ""
+    @State var amount = ""
+    @State var showAlert = false
+    @Binding var personIndex: Int
+    @Binding var dayIndex: Int
+    @Binding var positionIndex: Int
+    @ObservedObject var expensesStore: ExpensesStore
+    var operation: OperationType
+    @State var navigationText = ""
+    @State var actionButtonText = ""
+    
+    var body: some View {
+        NavigationView {
+            Form {
+                Section(header: Text("Name")) {
+                    TextField("Enter name here", text: $name)
+                }
+                Section(header: Text("Amount")) {
+                    TextField("Enter amount here", text: $amount)
+                }
+            }
+            .onAppear(perform: {
+                checkTranferedExpense()
+                checkOperationType(operation: operation)
+            })
+            .navigationBarTitle(navigationText, displayMode: .inline)
+            .navigationBarItems(
+                leading:
+                    Button(action: { expense = nil }) {
+                        Text("Cancel")
+                    },
+                trailing:
+                    Button(action: { handleExpense(operation: operation) }) {
+                        Text(actionButtonText)
+                    }
+                    .disabled(self.name.isEmpty || self.amount.isEmpty)
+                    .alert(isPresented: $showAlert) {
+                        Alert(title: Text("Information"), message: Text("Entered amount is incorrect"), dismissButton: .cancel(Text("OK")))
```

Comment:
for the test project this is fine, but in production apps you would want to have strings defined at the top of the file in some strings struct for localisation and convenience purposes.